sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/encryption/EncryptionProcessor.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.encryption;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.InternalServerErrorException;
+import com.azure.cosmos.implementation.RMResources;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.encryption.api.CosmosEncryptionAlgorithm;
+import com.azure.cosmos.implementation.encryption.api.DataEncryptionKey;
+import com.azure.cosmos.implementation.encryption.api.DataEncryptionKeyProvider;
+import com.azure.cosmos.implementation.encryption.api.EncryptionOptions;
+import com.azure.cosmos.implementation.guava27.Strings;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+public class EncryptionProcessor {
+
+    public ObjectNode encryptAsync(
+        ObjectNode itemJObj,
+        EncryptionOptions encryptionOptions,
+        DataEncryptionKeyProvider dataEncryptionKeyProvider) {
+        assert (itemJObj != null);
+        assert (encryptionOptions != null);
+        assert (encryptionOptions.getPathsToEncrypt() != null);
+        assert (!encryptionOptions.getPathsToEncrypt().isEmpty());
+
+        for (String path : encryptionOptions.getPathsToEncrypt()) {
+            if (StringUtils.isEmpty(path) || path.charAt(0) != '/' || path.lastIndexOf('/') != 0) {
+                throw new IllegalArgumentException("Invalid encryption path: " + path);
+            }
+        }
+
+        if (encryptionOptions.getDataEncryptionKeyId() == null) {
+            throw new IllegalArgumentException("Invalid encryption options: encryptionOptions.getDataEncryptionKeyId." + encryptionOptions.getDataEncryptionKeyId());
+        }
+
+        if (encryptionOptions.getEncryptionAlgorithm() == null) {
+            throw new IllegalArgumentException("Invalid encryption options: encryptionOptions.getEncryptionAlgorithm." + encryptionOptions.getEncryptionAlgorithm());
+        }
+
+        if (dataEncryptionKeyProvider == null) {
+            throw new IllegalArgumentException(RMResources.EncryptionKeyProviderNotConfigured);
+        }
+
+        ObjectNode toEncryptJObj = Utils.getSimpleObjectMapper().createObjectNode();
+
+        for (String pathToEncrypt : encryptionOptions.getPathsToEncrypt()) {
+            String propertyName = pathToEncrypt.substring(1);
```

Comment:
Does it work for nested JPath?

Reply:
for now it doesn't I am tracking this after this PR.